pkg/ctl/utils/install_corends.go

```diff
@@ -57,13 +55,6 @@ func doInstallCoreDNS(rc *cmdutils.ResourceCmd) error {
 		return errors.Wrapf(err, "getting credentials for cluster %q", meta.Name)
 	}
 
-	switch ctl.ControlPlaneVersion() {
-	case "":
```

Comment:
Are we sure about removing this?

Reply:
@2opremio said it was ok to do, we wanted you to tell us whether it was actually ok. Shall we still leave the case for "" ?

Reply:
I'll need to review what the function does.

Reply:
So the function currently relies on cache, it's possible for it to return an empty string.

https://github.com/weaveworks/eksctl/blob/079c5b71ee90b9fcfa6d4c10f07e81a2c8fc0ac0/pkg/eks/eks.go#L77-L83

The cache is populated when we make `ctl.GetCredentials` call, which is the one that we must do whenever we want to do anything with a cluster. Cache would be empty only if someone messes up the order of calls, or does something else accidentally.

https://github.com/weaveworks/eksctl/blob/079c5b71ee90b9fcfa6d4c10f07e81a2c8fc0ac0/pkg/eks/eks.go#L72

I think it's a minor detail that can improved, and we can open an issue for it if we'd like to discuss this, but thinking about it right now would be a distraction.

Let's put the empty string case back.